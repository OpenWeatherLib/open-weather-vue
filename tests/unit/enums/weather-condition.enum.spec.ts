import WeatherCondition from "@/enums/weather-condition.enum";

/*

describe("values", () => {
    test("should not change unexpected", () => {
        // Arrange + Act
        const actual: WeatherCondition[] = WeatherCondition.values;

        // Assert
        expect(actual).toMatchSnapshot([
            { id: 0, description: "Null", wallpaper: "/assets/weather_wallpaper_dummy.png", icon: "/assets/weather_dummy.png", count: 0 },
            { id: 1, description: "Clear", wallpaper: "/assets/weather_wallpaper_clear.png", icon: "/assets/weather_clear.png", count: 0 },
            { id: 2, description: "Clouds", wallpaper: "/assets/weather_wallpaper_cloud.png", icon: "/assets/weather_cloud.png", count: 0 },
            { id: 3, description: "Drizzle", wallpaper: "/assets/weather_wallpaper_drizzle.png", icon: "/assets/weather_drizzle.png", count: 0 },
            { id: 4, description: "Fog", wallpaper: "/assets/weather_wallpaper_fog.png", icon: "/assets/weather_fog.png", count: 0 },
            { id: 5, description: "Haze", wallpaper: "/assets/weather_wallpaper_haze.png", icon: "/assets/weather_haze.png", count: 0 },
            { id: 6, description: "Mist", wallpaper: "/assets/weather_wallpaper_mist.png", icon: "/assets/weather_mist.png", count: 0 },
            { id: 7, description: "Rain", wallpaper: "/assets/weather_wallpaper_rain.png", icon: "/assets/weather_rain.png", count: 0 },
            { id: 8, description: "Sleet", wallpaper: "/assets/weather_wallpaper_sleet.png", icon: "/assets/weather_sleet.png", count: 0 },
            { id: 9, description: "Snow", wallpaper: "/assets/weather_wallpaper_snow.png", icon: "/assets/weather_snow.png", count: 0 },
            { id: 10, description: "Squalls", wallpaper: "/assets/weather_wallpaper_squalls.png", icon: "/assets/weather_squalls.png", count: 0 },
            { id: 11, description: "Sun", wallpaper: "/assets/weather_wallpaper_sun.png", icon: "/assets/weather_sun.png", count: 0 },
            { id: 12, description: "Thunderstorm", wallpaper: "/assets/weather_wallpaper_thunderstorm.png", icon: "/assets/weather_thunderstorm.png", count: 0 }
        ]);
    });
});

describe("getByDescription", () => {
    test.each([
        [{ id: 0, description: "Null", wallpaper: "/assets/weather_wallpaper_dummy.png", icon: "/assets/weather_dummy.png", count: 0 }, "null"],
        [{ id: 1, description: "Clear", wallpaper: "/assets/weather_wallpaper_clear.png", icon: "/assets/weather_clear.png", count: 0 }, "clear"],
        [{ id: 2, description: "Clouds", wallpaper: "/assets/weather_wallpaper_cloud.png", icon: "/assets/weather_cloud.png", count: 0 }, "clouds"],
        [{ id: 3, description: "Drizzle", wallpaper: "/assets/weather_wallpaper_drizzle.png", icon: "/assets/weather_drizzle.png", count: 0 }, "drizzle"],
        [{ id: 4, description: "Fog", wallpaper: "/assets/weather_wallpaper_fog.png", icon: "/assets/weather_fog.png", count: 0 }, "fog"],
        [{ id: 5, description: "Haze", wallpaper: "/assets/weather_wallpaper_haze.png", icon: "/assets/weather_haze.png", count: 0 }, "haze"],
        [{ id: 6, description: "Mist", wallpaper: "/assets/weather_wallpaper_mist.png", icon: "/assets/weather_mist.png", count: 0 }, "mist"],
        [{ id: 7, description: "Rain", wallpaper: "/assets/weather_wallpaper_rain.png", icon: "/assets/weather_rain.png", count: 0 }, "rain"],
        [{ id: 8, description: "Sleet", wallpaper: "/assets/weather_wallpaper_sleet.png", icon: "/assets/weather_sleet.png", count: 0 }, "sleet"],
        [{ id: 9, description: "Snow", wallpaper: "/assets/weather_wallpaper_snow.png", icon: "/assets/weather_snow.png", count: 0 }, "snow"],
        [{ id: 10, description: "Squalls", wallpaper: "/assets/weather_wallpaper_squalls.png", icon: "/assets/weather_squalls.png", count: 0 }, "squalls"],
        [{ id: 11, description: "Sun", wallpaper: "/assets/weather_wallpaper_sun.png", icon: "/assets/weather_sun.png", count: 0 }, "sun"],
        [{ id: 12, description: "Thunderstorm", wallpaper: "/assets/weather_wallpaper_thunderstorm.png", icon: "/assets/weather_thunderstorm.png", count: 0 }, "thunderstorm"],
        [{ id: 0, description: "Null", wallpaper: "/assets/weather_wallpaper_dummy.png", icon: "/assets/weather_dummy.png", count: 0 }, "fake"],
        [{ id: 0, description: "Null", wallpaper: "/assets/weather_wallpaper_dummy.png", icon: "/assets/weather_dummy.png", count: 0 }, undefined]
    ])("should return WeatherCondition %s for description %s", (expected: WeatherCondition, description: string) => {
        // Arrange & Act
        const actual: WeatherCondition = WeatherCondition.getByDescription(description);

        // Assert
        expect(actual).toStrictEqual(expected);
    })
});

*/
