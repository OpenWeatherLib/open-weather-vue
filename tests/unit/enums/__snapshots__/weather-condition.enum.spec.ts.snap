// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`values should not change unexpected 1`] = `
Object {
  "0": Object {
    "count": 0,
    "description": "Null",
    "icon": "/assets/weather_dummy.png",
    "id": 0,
    "wallpaper": "/assets/weather_wallpaper_dummy.png",
  },
  "1": Object {
    "count": 0,
    "description": "Clear",
    "icon": "/assets/weather_clear.png",
    "id": 1,
    "wallpaper": "/assets/weather_wallpaper_clear.png",
  },
  "10": Object {
    "count": 0,
    "description": "Squalls",
    "icon": "/assets/weather_squalls.png",
    "id": 10,
    "wallpaper": "/assets/weather_wallpaper_squalls.png",
  },
  "11": Object {
    "count": 0,
    "description": "Sun",
    "icon": "/assets/weather_sun.png",
    "id": 11,
    "wallpaper": "/assets/weather_wallpaper_sun.png",
  },
  "12": Object {
    "count": 0,
    "description": "Thunderstorm",
    "icon": "/assets/weather_thunderstorm.png",
    "id": 12,
    "wallpaper": "/assets/weather_wallpaper_thunderstorm.png",
  },
  "2": Object {
    "count": 0,
    "description": "Clouds",
    "icon": "/assets/weather_cloud.png",
    "id": 2,
    "wallpaper": "/assets/weather_wallpaper_cloud.png",
  },
  "3": Object {
    "count": 0,
    "description": "Drizzle",
    "icon": "/assets/weather_drizzle.png",
    "id": 3,
    "wallpaper": "/assets/weather_wallpaper_drizzle.png",
  },
  "4": Object {
    "count": 0,
    "description": "Fog",
    "icon": "/assets/weather_fog.png",
    "id": 4,
    "wallpaper": "/assets/weather_wallpaper_fog.png",
  },
  "5": Object {
    "count": 0,
    "description": "Haze",
    "icon": "/assets/weather_haze.png",
    "id": 5,
    "wallpaper": "/assets/weather_wallpaper_haze.png",
  },
  "6": Object {
    "count": 0,
    "description": "Mist",
    "icon": "/assets/weather_mist.png",
    "id": 6,
    "wallpaper": "/assets/weather_wallpaper_mist.png",
  },
  "7": Object {
    "count": 0,
    "description": "Rain",
    "icon": "/assets/weather_rain.png",
    "id": 7,
    "wallpaper": "/assets/weather_wallpaper_rain.png",
  },
  "8": Object {
    "count": 0,
    "description": "Sleet",
    "icon": "/assets/weather_sleet.png",
    "id": 8,
    "wallpaper": "/assets/weather_wallpaper_sleet.png",
  },
  "9": Object {
    "count": 0,
    "description": "Snow",
    "icon": "/assets/weather_snow.png",
    "id": 9,
    "wallpaper": "/assets/weather_wallpaper_snow.png",
  },
}
`;
